packages/fuels-programs/src/contract.rs

```diff
@@ -186,6 +187,7 @@ impl Contract {
         wallet: &WalletUnlocked,
         params: TxParameters,
         storage_configuration: StorageConfiguration,
+        configurables: Configurables,
```

Comment:
Deploying is starting to have way too many parameters. 

We should probably group them inside a `DeployConfig` with sensible defaults and a builder-like interface so that we may nuke the two deploy methods in favor of one with this structure.

Wdyt?

Reply:
I like the idea. What do you think we make a new issue for this ?

Reply:
Sure. Maybe not to put all the current deploy fn arguments in one struct, maybe only to group the params and various configurations (the oft-defaultable part) and let the others remain as arguments. 

In any case a new issue is a good idea, the implementer can decide on the best course of action.

Reply:
Ha! I _just_ thought the same thing; it's getting way too complicated. But yeah, a separate issue/PR for this!

Reply:
https://github.com/FuelLabs/fuels-rs/issues/855